### PR TITLE
Improve docs and implementation of `getBindingTargets` method

### PR DESCRIPTION
**Pull request by ma2ciek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Internal: Improved docs and implementation of `getBindingTargets` method.

### CODE REVIEW

File: src/observablemixin.js [hunk 1]
```diff
@@ -445,9 +445,12 @@ function bindToMany( observables, attribute, callback ) {
 //
 // @param {Iterable.<Observable>} observables
 // @param {String} attribute
-// @returns {Array.<String>}
+// @returns {Array.<String|Observable>}
 function getBindingTargets( observables, attribute ) {
-	return Array.prototype.concat( ...observables.map( observable => [ observable, attribute ] ) );
+	const observableAndAttributePairs = Array.from( observables ).map( observable => [ observable, attribute ] );
```
Comment (Reinmar) [inline src/observablemixin.js]: Why do you change implementation? Is support for iterables needed?

Comment (ma2ciek) [reply src/observablemixin.js]: I can revert it and allow only arrays here.

But there's the second change I've made. I changed `Array.prototype.concat()` to `Array.prototype.concat.apply()`. It looked strange to me, that `this` refers to `undefined` here.

Comment (Reinmar) [reply src/observablemixin.js]: It's not `undefined` – it equalled `Array.prototype` ;) And yes – it was a bit ugly and you can change it.

PS. Please remember about updating the docs after removing `Array.from()` call.

Comment (Reinmar) [reply src/observablemixin.js]: PPS. There would be nothing wrong with supporting iterables if that was required by the public API. This is a private function so it should do only what it's needed from it.

Comment (ma2ciek) [reply src/observablemixin.js]: It reflects also on the `bindToMany` function and the exposed `bind().toMany()`. There's no need for supporting iterables in the exposed function I suppose, so I'll change it.

**Comment by ma2ciek** [comment]
Fixed.

**Comment by Reinmar** [comment]
BTW, the correct msg was:

> Internal: Improved docs and implementation of the `getBindingTargets()` function.

**Pull request merged**